src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsResource.cs

```diff
@@ -45,10 +45,43 @@ public class AzureEventHubsResource(string name, Action<AzureResourceInfrastruct
     /// <summary>
     /// Gets the connection string template for the manifest for the Azure Event Hubs endpoint.
     /// </summary>
-    public ReferenceExpression ConnectionStringExpression =>
-        IsEmulator
-        ? ReferenceExpression.Create($"Endpoint=sb://{EmulatorEndpoint.Property(EndpointProperty.Host)}:{EmulatorEndpoint.Property(EndpointProperty.Port)};SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=SAS_KEY_VALUE;UseDevelopmentEmulator=true;")
-        : ReferenceExpression.Create($"{EventHubsEndpoint}");
+    public ReferenceExpression ConnectionStringExpression => BuildConnectionString();
+
+    private ReferenceExpression BuildConnectionString()
+    {
+        var builder = new ReferenceExpressionBuilder();
+
+        if (IsEmulator)
+        {
+            // ConnectionString: Endpoint=...
+            builder.Append($"Endpoint=sb://{EmulatorEndpoint.Property(EndpointProperty.Host)}:{EmulatorEndpoint.Property(EndpointProperty.Port)};SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=SAS_KEY_VALUE;UseDevelopmentEmulator=true");
+        }
+        else
+        {
+            // FQNS: Uri format, e.g. https://...
+            builder.Append($"{EventHubsEndpoint}");
+        }
+
+        if (!Hubs.Any(hub => hub.IsDefaultEntity))
+        {
+            // Of zero or more hubs, none are flagged as default
+            return builder.Build();
+        }
+
+        // Of one or more hubs, only one may be flagged as default
+        var defaultEntity = Hubs.Single(hub => hub.IsDefaultEntity);
+
+        if (IsEmulator)
+        {
+            builder.Append($";EntityPath={defaultEntity.Name}");
+        }
+        else
+        {
+            builder.Append($"?EntityPath={defaultEntity.Name}");
```

Comment:
How does the Azure.Messaging.EventHubs library handle having this query on the Uri?

Reply:
It doesn't care, which is why I ab(use) it.

Reply:
Looks like this was explicitly implemented in https://github.com/Azure/azure-sdk-for-net/pull/41819. FYI - @jsquire in case this wasn't intended to be supported as part of that change. .NET Aspire is going to depend on the Query of the Uri being ignored (or at least not throwing an exception).